server/src/main/java/org/exoplatform/chat/services/mongodb/ChatServiceImpl.java

```diff
@@ -259,7 +261,9 @@ public String read(String room, UserService userService, boolean isTextOnly, Lon
     {
       tsobj.append("$lt", toTimestamp);
     }
-    query.put("timestamp", tsobj);
+    BasicDBObject ts = new BasicDBObject("timestamp",tsobj);
+    BasicDBObject updts = new BasicDBObject("lastUpdatedTimestamp",tsobj);
```

Comment:
Yes probably, I dont know performance impact as large majority of messages have no lastUpdatedTimestamp. i'll do a perf test to confirm